scapy/fields.py

```diff
@@ -274,7 +274,7 @@ def __findaddr(self, pkt):
         dst = ("0.0.0.0" if self.dstname is None
                else getattr(pkt, self.dstname))
         if isinstance(dst, (Gen, list)):
-            r = {conf.route.route(daddr) for daddr in dst}
+            r = {conf.route.route(str(daddr)) for daddr in dst}
```

Comment:
Those will convert potential Net(/Net6 for the other one) generators

Reply:
I don't understand. Could you show something that wouldn't work without your patch and you are fixing?

Reply:
Sorry ! (those were provided in the linked issue).
I'll add a better PoC